lib/msal-common/src/telemetry/performance/PerformanceEvent.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export enum PerformanceEvents {
+    
+    /**
+     * acquireTokenByCode API (msal-browser and msal-node). 
+     * Used to acquire tokens by trading an authorization code against the token endpoint.
+     */
+    AcquireTokenByCode = "acquireTokenByCode",
+    
+    /**
+     * acquireTokenByRefreshToken API (msal-browser and msal-node).
+     * Used to renew an access token using a refresh token against the token endpoint.
+     */
+    AcquireTokenByRefreshToken = "acquireTokenByRefreshToken",
+    
+    /**
+     * acquireTokenSilent API (msal-browser and msal-node).
+     * Used to silently acquire a new access token (from the cache or the network).
+     */
+    AcquireTokenSilent = "acquireTokenSilent",
+    
+    /**
+     * acquireTokenSilentAsync (msal-browser).
+     * Internal API for acquireTokenSilent.
+     */
+    AcquireTokenSilentAsync = "acquireTokenSilentAsync",
+    
+    /**
+     * getPublicKeyThumbprint API in CryptoOpts class (msal-browser).
+     * Used to generate a public/private keypair and generate a public key thumbprint for pop requests.
+     */
+    CryptoOptsGetPublicKeyThumbprint = "cryptoOptsGetPublicKeyThumbprint",
+    
+    /**
+     * signJwt API in CryptoOpts class (msal-browser).
+     * Used to signed a pop token.
+     */
+    CryptoOptsSignJwt = "cryptoOptsSignJwt",
+    
+    /**
+     * acquireToken API in the SilentCacheClient class (msal-browser).
+     * Used to read access tokens from the cache.
+     */
+    SilentCacheClientAcquireToken = "silentCacheClientAcquireToken",
```

Comment:
If we ever decide to reorganize or rename our internal APIs, will that cause a problem here?

Reply:
Yes, this is in theory a potential issue. This can be handled in reporting by limiting relevant queries to certain versions of the browser.

Reply:
The strings are ultimately arbitrary, so its up to us if we keep a given string after a refactor, or change it and update reporting accordingly.